src/main/java/ch/njol/skript/expressions/ExprOptions.java

```diff
@@ -0,0 +1,129 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.expressions;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.bukkit.event.Event;
+import org.eclipse.jdt.annotation.Nullable;
+import org.skriptlang.skript.lang.script.Script;
+
+import com.google.common.collect.Lists;
+
+import ch.njol.skript.ScriptLoader;
+import ch.njol.skript.Skript;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.util.SimpleExpression;
+import ch.njol.skript.structures.StructOptions.OptionsData;
+import ch.njol.util.Kleenean;
+
+@Name("Script Options")
+@Description("Returns one or more options from any script. If no script is provided, it will be the current script of the syntax.")
+@Examples({
+	"# example.sk",
+	"options:",
+		"\texample: Hello World",
+	"",
+	"command /example <string=%script option \"test\"%>:",
+		"\ttrigger:",
+			"\t\tsend \"The option is %script option \"\"example\"\"%\"",
+	"",
+	"# this.sk",
+	"options:",
+		"\texample 2: 1337",
+	"command /anotherFile:",
+		"\ttrigger:",
+			"\t\tsend \"The options are %script options \"\"example 2\"\" and \"\"example\"\" from scripts \"\"this.sk\"\" and \"\"example.sk\"\"%\""
+})
+@Since("INSERT VERSION")
+public class ExprOptions extends SimpleExpression<String> {
+
+	static {
+		Skript.registerExpression(ExprOptions.class, String.class, ExpressionType.COMBINED, "[the] [s(c|k)ript] option[s] %strings% [(from|in) [script[s]] %-strings%]");
+	}
+
+	@Nullable
+	private Expression<String> scripts;
+	private Expression<String> options;
+	private Script current;
+
+	@Override
+	@SuppressWarnings("unchecked")
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		current = getParser().getCurrentScript();
```

Comment:
Causes issues when run as an effect command: `send option "e"`
```
[17:14:58 ERROR]: #!#!
[17:14:58 ERROR]: #!#! [Skript] Severe Error:
[17:14:58 ERROR]: #!#! Unexpected error while executing effect command 'send option "e"' by 'Sahvde'
[17:14:58 ERROR]: #!#!
[17:14:58 ERROR]: #!#! You're running a (buggy) nightly version of Skript.
[17:14:58 ERROR]: #!#! If this is not a test server, switch to a more stable release NOW!
[17:14:58 ERROR]: #!#! Your players are unlikely to appreciate crashes and/or data loss due to Skript bugs.
[17:14:58 ERROR]: #!#!
[17:14:58 ERROR]: #!#! Just testing things? Good. Please report this bug, so that we can fix it before a stable release.[17:14:58 ERROR]: #!#! Issue tracker: https://github.com/SkriptLang/Skript/issues
[17:14:58 ERROR]: #!#!
[17:14:58 ERROR]: #!#! Stack trace:
[17:14:58 ERROR]: #!#! ch.njol.skript.SkriptAPIException: This ParserInstance is not currently parsing/loading something!
[17:14:58 ERROR]: #!#!     at Skript-nightly.jar//ch.njol.skript.lang.parser.ParserInstance.getCurrentScript(ParserInstance.java:129)
[17:14:58 ERROR]: #!#!     at Skript-nightly.jar//ch.njol.skript.expressions.ExprOptions.init(ExprOptions.java:80)
[17:14:58 ERROR]: #!#!     at Skript-nightly.jar//ch.njol.skript.lang.SkriptParser.parse(SkriptParser.java:266)
[17:14:58 ERROR]: #!#!     at Skript-nightly.jar//ch.njol.skript.lang.SkriptParser.parse(SkriptParser.java:176)
[17:14:58 ERROR]: #!#!     at Skript-nightly.jar//ch.njol.skript.lang.SkriptParser.parseSingleExpr(SkriptParser.java:531)
[17:14:58 ERROR]: #!#!     at Skript-nightly.jar//ch.njol.skript.lang.SkriptParser.parseExpression(SkriptParser.java:761)
[17:14:58 ERROR]: #!#!     at Skript-nightly.jar//ch.njol.skript.patterns.TypePatternElement.match(TypePatternElement.java:154)
[17:14:58 ERROR]: #!#!     at Skript-nightly.jar//ch.njol.skript.patterns.PatternElement.matchNext(PatternElement.java:54)
[17:14:58 ERROR]: #!#!     at Skript-nightly.jar//ch.njol.skript.patterns.LiteralPatternElement.match(LiteralPatternElement.java:65)
[17:14:58 ERROR]: #!#!     at Skript-nightly.jar//ch.njol.skript.patterns.PatternElement.matchNext(PatternElement.java:54)
[17:14:58 ERROR]: #!#!     at Skript-nightly.jar//ch.njol.skript.patterns.OptionalPatternElement.match(OptionalPatternElement.java:46)
[17:14:58 ERROR]: #!#!     at Skript-nightly.jar//ch.njol.skript.patterns.PatternElement.matchNext(PatternElement.java:54)
[17:14:58 ERROR]: #!#!     at Skript-nightly.jar//ch.njol.skript.patterns.LiteralPatternElement.match(LiteralPatternElement.java:65)
[17:14:58 ERROR]: #!#!     at Skript-nightly.jar//ch.njol.skript.patterns.ChoicePatternElement.match(ChoicePatternElement.java:63)
[17:14:58 ERROR]: #!#!     at Skript-nightly.jar//ch.njol.skript.patterns.GroupPatternElement.match(GroupPatternElement.java:47)
[17:14:58 ERROR]: #!#!     at Skript-nightly.jar//ch.njol.skript.patterns.SkriptPattern.match(SkriptPattern.java:58)
[17:14:58 ERROR]: #!#!     at Skript-nightly.jar//ch.njol.skript.lang.SkriptParser.parse_i(SkriptParser.java:1275)
[17:14:58 ERROR]: #!#!     at Skript-nightly.jar//ch.njol.skript.lang.SkriptParser.parse(SkriptParser.java:229)
[17:14:58 ERROR]: #!#!     at Skript-nightly.jar//ch.njol.skript.lang.SkriptParser.parse(SkriptParser.java:176)
[17:14:58 ERROR]: #!#!     at Skript-nightly.jar//ch.njol.skript.lang.Effect.parse(Effect.java:76)
[17:14:58 ERROR]: #!#!     at Skript-nightly.jar//ch.njol.skript.command.Commands.handleEffectCommand(Commands.java:212)[17:14:58 ERROR]: #!#!     at Skript-nightly.jar//ch.njol.skript.command.Commands$2$1.call(Commands.java:316)
[17:14:58 ERROR]: #!#!     at Skript-nightly.jar//ch.njol.skript.command.Commands$2$1.call(Commands.java:313)
[17:14:58 ERROR]: #!#!     at org.bukkit.craftbukkit.v1_20_R1.scheduler.CraftFuture.run(CraftFuture.java:88)
[17:14:58 ERROR]: #!#!     at org.bukkit.craftbukkit.v1_20_R1.scheduler.CraftScheduler.mainThreadHeartbeat(CraftScheduler.java:480)
[17:14:58 ERROR]: #!#!     at net.minecraft.server.MinecraftServer.b(MinecraftServer.java:1480)
[17:14:58 ERROR]: #!#!     at net.minecraft.server.dedicated.DedicatedServer.b(DedicatedServer.java:447)
[17:14:58 ERROR]: #!#!     at net.minecraft.server.MinecraftServer.a(MinecraftServer.java:1394)
[17:14:58 ERROR]: #!#!     at net.minecraft.server.MinecraftServer.w(MinecraftServer.java:1171)
[17:14:58 ERROR]: #!#!     at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:318)
[17:14:58 ERROR]: #!#!     at java.base/java.lang.Thread.run(Thread.java:833)
[17:14:58 ERROR]: #!#!
[17:14:58 ERROR]: #!#! Version Information:
[17:14:58 ERROR]: #!#!   Skript: 2.8.0-dev-nightly-06573ef
[17:14:58 ERROR]: #!#!     Flavor: skriptlang-nightly
[17:14:58 ERROR]: #!#!     Date: 17:07:24.625846788
[17:14:58 ERROR]: #!#!   Bukkit: 1.20-R0.1-SNAPSHOT
[17:14:58 ERROR]: #!#!   Minecraft: 1.20
[17:14:58 ERROR]: #!#!   Java: 17.0.4 (OpenJDK 64-Bit Server VM 17.0.4+8-LTS)
[17:14:58 ERROR]: #!#!   OS: Windows 10 amd64 10.0
[17:14:58 ERROR]: #!#!
[17:14:58 ERROR]: #!#! Server platform: Paper
[17:14:58 ERROR]: #!#!
[17:14:58 ERROR]: #!#! Current node: null
[17:14:58 ERROR]: #!#! Current item: null
[17:14:58 ERROR]: #!#!
[17:14:58 ERROR]: #!#! Thread: Server thread
[17:14:58 ERROR]: #!#!
[17:14:58 ERROR]: #!#! Language: english
[17:14:58 ERROR]: #!#! Link parse mode: DISABLED
[17:14:58 ERROR]: #!#!
[17:14:58 ERROR]: #!#! End of Error.
[17:14:58 ERROR]: #!#!
```